index.js

```diff
@@ -1,127 +1,326 @@
+          obje.value = param.value;
+          if (!param.enabled) {
+            obje.disabled = true;
+          } else {
+            str = str + obje.key + '=' + obje.value + '&';
+          }
+           item.request.url.query.push(obje);
+        });
+      }
+      item.request.url.raw = str;
+    }
+
+    
+
+    
+
+    _.each(dhcRequest.headers, function(dhcHeader) {
+      var header = {};
+      header.key = dhcHeader.name;
+      header.value = dhcHeader.value;
+      if (dhcHeader.enabled === false) {
+        header.disabled = true;
+      }
+      item.request.header.push(header);
+    });
+
+    if (this.methodsWithBody.indexOf(item.request.method.toUpperCase()) > -1) {
+      var dhcBody = dhcRequest.body;
+      var request = item.request.body; //here request refers to body.
+      if (
+        dhcBody.bodyType === 'Form' &&
+        dhcBody.formBody &&
+        dhcBody.formBody.encoding === 'multipart/form-data'
+      ) {
+        // multipart
+        request.mode = 'Formdata';
+        request.formdata = [];
+        _.each(dhcBody.formBody.items, function(item) {
+          if (item.type === 'Text') {
+            var requestdata = {};
+            requestdata.key = item.name;
+            requestdata.value = item.value;
+            requestdata.type = 'text';
+            if (!item.enabled) {
+              requestdata.disabled = true;
+            }
+            request.formdata.push(requestdata);
+          }
+        });
+      } else if (
+        dhcBody.bodyType === 'Form' &&
+        dhcBody.formBody &&
+        dhcBody.formBody.encoding === 'application/x-www-form-urlencoded'
+      ) {
+        // urlencoded
+        request.mode = 'urlencoded';
+        request.urlencoded = [];
+        _.each(dhcBody.formBody.items, function(item) {
+          if (item.type === 'Text') {
+            var requestdata = {};
+            requestdata.key = item.name;
+            requestdata.value = item.value;
+            requestdata.type = 'text';
+            if (!item.enabled) {
+              requestdata.disabled = true;
+            }
+            request.urlencoded.push(requestdata);
+          }
+        });
+      } else if (dhcBody.bodyType === 'Text' && dhcBody.textBody) {
+        // raw
+        request.mode = 'raw';
+        request.raw = dhcBody.textBody;
+      }
+    }
+    return item;
+  },
+
+  convertDhcProject: function(obj) {
+    var rootNode = _.find(obj.nodes, function(node) {
+        return node.type === 'Project';
+      }),
+      collection = this.createCollection(rootNode.name),
+      dhcServices = _.filter(obj.nodes, function(node) {
+        return node.type === 'Service';
+      }),
+      dhcScenarios = _.filter(obj.nodes, function(node) {
+        return node.type === 'Scenario';
+      }),
+      dhcRequests = _.filter(obj.nodes, function(node) {
+        return node.type === 'Request';
+      }),
+      oldThis = this;
+    //services
+    _.each(dhcServices, function(dhcService) {
```

Comment:
update all of this. A bottom up approach will require you to loop through each entity only once